### ISSUE TITLE
SVui Errors from log in of Beta

### ISSUE THREAD

**Issue by finalsliver** [FailcoderAddons/supervillain-ui#198]
Click arrowed lines for collapsable error from Swatter!
https://github.com/FailcoderAddons/supervillain-ui/blob/3f80381d47277be272a916598dd803aac4c371b6/SVUI_!Core/setup/presets.lua#L928
<details>
  <summary>SVUI_!Core\setup\presets.lua line 928:
   attempt to concatenate global 'ACTIONBAR_LABEL' (a nil value)</summary><p>

```lua
Date: 2018-06-28 13:06:47
ID: 1
Error occured in: Global
Count: 1
Message: ..\AddOns\SVUI_!Core\setup\presets.lua line 928:
   attempt to concatenate global 'ACTIONBAR_LABEL' (a nil value)
Debug:
   SVUI_!Core\setup\presets.lua:928:
      SVUI_!Core\setup\presets.lua:860
   SVUI_!Core\setup\presets.lua:1098: CopyPage()
   SVUI_!Core\setup\installer.lua:786: SetPage()
   SVUI_!Core\setup\installer.lua:1238: LoadInstaller()
   SVUI_!Core\setup\installer.lua:872: Install()
   SVUI_!Core\system\core.lua:745: VersionCheck()
   SVUI_!Core\system\core.lua:869: Initialize()
   ...erface\AddOns\SVUI_!Core\libs\_SVUI_Lib\Registry.lua:896:
      ...erface\AddOns\SVUI_!Core\libs\_SVUI_Lib\Registry.lua:876
Locals:
(*temporary) = <table> {
}
(*temporary) = <table> {
 Desc3 = "Choose Your Destiny OR Hit Continue"
 SubTitle = "This is SVUI version 1.4.2!"
 Desc1 = "Before I can turn you loose, persuing whatever villainy you feel will advance your professional career... I need to ask some questions and turn a few screws first."
 Desc2 = "At any time you can get to the config options by typing the command  / sv. For quick changes to frame, bar or color sets, call your henchman by clicking the button on the bottom right of your screen. (Its the one with his stupid face on it)"
 Option02 = <table> {
 }
 Option01 = <table> {
 }
}
(*temporary) = <table> {
 Desc3 = "Choose Your Destiny OR Hit Continue"
 SubTitle = "Chat"
 Desc1 = "Whether you want to or not, you will be needing a communicator so other villains can either update you on their doings-of-evil or inform you about the MANY abilities of Chuck Norris"
 Desc2 = "The chat windows function the same as standard chat windows, you can right click the tabs and drag them, rename them, slap them around, you know... whatever. Clickity-click to setup your chat windows."
 Option1 = <table> {
 }
}
(*temporary) = <table> {
 Desc3 = "Choose Your Destiny OR Hit Continue"
 SubTitle = "Resolution"
 Desc1 = "Your current resolution is 1920x1080, this is considered a High resolution."
 Desc2 = <function> defined @Interface\AddOns\SVUI_!Core\setup\presets.lua:66
 Option2 = <table> {
 }
 Option1 = <table> {
 }
}
(*temporary) = <table> {
}
(*temporary) = <table> {
 Desc3 = "Choose Your Destiny OR Hit Continue"
 SubTitle = "Unit Frames Settings"
 Desc1 = "You can now choose what primary unitframe style you wish to use."
 Desc2 = "This will change the layout of your unitframes (ie.. Player, Target, Pet, Party, Raid ...etc)."
 Option1 = <table> {
 }
 REQUIRED = "UnitFrames"
 Option2 = <table> {
 }
 Option3 = <table> {
 }
}
(*temporary) = <table> {
 REQUIRED = "UnitFrames"
 Desc3 = "Choose Your Destiny OR Hit Continue"
 SubTitle = "Group Layout"
 Desc1 = "You can now choose what group layout you prefer."
 Desc2 = "This will adjust various settings on group units, attempting to make certain roles more usable"
 Option1 = <table> {
 }
 Option4 = <table> {
 }
 Option2 = <table> {
 }
 Option3 = <table> {
 }
}
(*temporary) = <table> {
 REQUIRED = "ActionBars"
}
(*temporary) = nil
(*temporary) = " Settings"
(*temporary) = "Settings"
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to concatenate global 'ACTIONBAR_LABEL' (a nil value)"
PAGE_DIALOG = nil
L = <table> {
 Informative Panels = "Informative Panels"
 Direct AoE heals will let the unit frames of the affected party / raid members glow for the defined time period. = "Direct AoE heals will let the unit frames of the affected party / raid members glow for the defined time period."
 Stack Bank to Bags = "Stack Bank to Bags"
 Bags = "Bags"
 At any time you can get to the config options by typing the command /sv. For quick changes to frame, bar or color sets, call your henchman by clicking the button on the bottom right of your screen. (Its the one with his stupid face on it) = "At any time you can get to the config options by typing the command /sv. For quick changes to frame, bar or color sets, call your henchman by clicking the button on the bottom right of your screen. (Its the one with his stupid face on it)"
 This theme tells the world that you are a villain who can put on a show = "This theme tells the world that you are a villain who can put on a show"
 I will attempt to preserve some of your basic settings but no promises. This will clean out everything else. Are you sure you want to reset everything? = "I will attempt to preserve some of your basic settings but no promises. This will clean out everything else. Are you sure you want to reset everything?"
 Avoidance Breakdown = "Avoidance Breakdown"
 Raid = "Raid"
 Choose a layout for your action bars. = "Choose a layout for your action bars."
 yells = "yells"
 Compact = "Compact"
 The better to PEW-PEW you with my dear! = "The better to PEW-PEW you with my dear!"
 Your items have been repair
AddOns:
  Swatter, v7.3.1 (<%codename%>)
  SVUICore, v1.4.2
  BlizRuntimeLib_enUS v8.0.1.80000 <none>
  (ck=5e)
```
</p>
</details>

---

https://github.com/FailcoderAddons/supervillain-ui/blob/3f80381d47277be272a916598dd803aac4c371b6/SVUI_!Core/libs/LibArtifactData-1.0/LibArtifactData-1.0.lua#L212-L221
<details>
<summary>\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua line 213:
   attempt to call upvalue 'GetArtifactKnowledgeLevel' (a nil value)</summary><p>

```lua
Date: 2018-06-28 13:06:48
ID: 2
Error occured in: Global
Count: 429
Message: ...ore\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua line 213:
   attempt to call upvalue 'GetArtifactKnowledgeLevel' (a nil value)
Debug:
   ...ore\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:213:
      ...ore\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:212
   ...ore\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:224:
      ...ore\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:223
   ...ore\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:262:
      ...ore\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:254
   ...ore\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:273:
      ...ore\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:270
   ...ore\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:314: ?()
   ...ore\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:68:
      ...ore\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:68
Locals:
(*temporary) = nil
(*temporary) = "attempt to call upvalue 'GetArtifactKnowledgeLevel' (a nil value)"
GetArtifactKnowledgeLevel = nil
GetArtifactKnowledgeMultiplier = nil
artifacts = <table> {
 knowledgeMultiplier = 1
 knowledgeLevel = 0
}
Debug = <function> defined @Interface\AddOns\SVUI_!Core\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:9
lib = <table> {
 GetActiveArtifactID = <function> defined @Interface\AddOns\SVUI_!Core\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:425
 GetArtifactKnowledge = <function> defined @Interface\AddOns\SVUI_!Core\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:476
 ForceUpdate = <function> defined @Interface\AddOns\SVUI_!Core\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:509
 callbacks = <table> {
 }
 GetAcquiredArtifactPower = <function> defined @Interface\AddOns\SVUI_!Core\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:480
 GetArtifactRelics = <function> defined @Interface\AddOns\SVUI_!Core\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:458
 RegisterCallback = <function> defined @Interface\AddOns\SVUI_!Core\libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:116
 GetArtifactPower = <function> defined @Interface\AddOns\SVUI_!Core\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:467
 GetArtifactTraits = <function> defined @Interface\AddOns\SVUI_!Core\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:449
 UnregisterCallback = <function> defined @Interface\AddOns\SVUI_!Core\libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:179
 GetArtifactInfo = <function> defined @Interface\AddOns\SVUI_!Core\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:429
 GetNumObtainedArtifacts = <function> defined @Interface\AddOns\SVUI_!Core\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:438
 frame = <unnamed> {
 }
 GetAllArtifactsInfo = <function> defined @Interface\AddOns\SVUI_!Core\libs\LibArtifactData-1.0\LibArtifactData-1.0.lua:434
 UnregisterAllCallbacks = <function> defined @Interface\AddOns\SVUI_!Core\libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:200
}

AddOns:
  Swatter, v7.3.1 (<%codename%>)
  SVUICore, v1.4.2
  BlizRuntimeLib_enUS v8.0.1.80000 <none>
  (ck=5e)


```

</p>
</details>

---
https://github.com/FailcoderAddons/supervillain-ui/blob/3f80381d47277be272a916598dd803aac4c371b6/SVUI_Auras/SVUI_Auras.lua#L189

<details>
<summary>SVUI_Auras\SVUI_Auras.lua line 189:
   attempt to compare number with nil</summary><p>

```lua
Date: 2018-06-28 13:38:27
ID: 3
Error occured in: Global
Count: 333
Message: ..\AddOns\SVUI_Auras\SVUI_Auras.lua line 189:
   attempt to compare number with nil
Debug:
   SVUI_Auras\SVUI_Auras.lua:189:
      SVUI_Auras\SVUI_Auras.lua:168
   [C]: SetAttribute()
   ..\FrameXML\SecureGroupHeaders.lua:760:
      ..\FrameXML\SecureGroupHeaders.lua:728
   ..\FrameXML\SecureGroupHeaders.lua:1046: SecureAuraHeader_Update()
   ..\FrameXML\SecureGroupHeaders.lua:686:
      ..\FrameXML\SecureGroupHeaders.lua:682
Locals:
self = SVUI_PlayerBuffsAuraButton1 {
 0 = <userdata>
 count = <unnamed> {
 }
 time = <unnamed> {
 }
 highlight = <unnamed> {
 }
 anim = Flash {
 }
 texture = <unnamed> {
 }
}
attribute = "index"
auraIndex = 1
filter = "HELPFUL"
unit = "player"
name = "Beezil's Cog"
_ = 237293
icon = 0
count = nil
dispelType = 0
val = 0
expires = "player"
caster = false
(*temporary) = <function> defined =[C]:-1
(*temporary) = SVUI_PlayerBuffsAuraButton1 {
 0 = <userdata>
 count = <unnamed> {
 }
 time = <unnamed> {
 }
 highlight = <unnamed> {
 }
 anim = Flash {
 }
 texture = <unnamed> {
 }
}
(*temporary) = "OnUpdate"
(*temporary) = nil
(*temporary) = <userdata>
(*temporary) = 85619
(*temporary) = "attempt to compare number with nil"
UnitAura = <function> defined =[C]:-1
GetTime = <function> defined =[C]:-1
RefreshAuraTime = <function> defined @Interface\AddOns\SVUI_Auras\SVUI_Auras.lua:114
GetInventoryItemQuality = <function> defined =[C]:-1
GetInventoryItemTexture = <function> defined =[C]:-1
GetItemQualityColor = <function> defined =[C]:-1
select = <function> defined =[C]:-1
GetWeaponEnchantInfo = <function> defined =[C]:-1

AddOns:
  Swatter, v7.3.1 (<%codename%>)
  SVUICore, v1.4.2
  SVUIActionBars, v1.4.2
  SVUIAuras, v1.4.2
  SVUIChat, v1.4.2
  SVUICraftOMatic, v1.4.2
  SVUIFightOMatic, v1.4.2
  SVUIInventory, v1.4.2
  SVUIMaps, v1.4.2
  SVUINamePlates, v1.4.2
  SVUIQuestTracker, v1.4.2
  SVUISkins, v1.4.2
  SVUITooltip, v1.4.2
  SVUIUnitFrames, v1.4.2
  BlizRuntimeLib_enUS v8.0.1.80000 <none>
  (ck=174)


```

</p>
</details>

---
https://github.com/FailcoderAddons/supervillain-ui/blob/3f80381d47277be272a916598dd803aac4c371b6/SVUI_Skins/components/blizzard/_load.xml#L2

<details>
<summary>Vanilla WoW Lua Warning 1</summary><p>

```
Message: Error loading Interface\AddOns\SVUI_Skins\components\blizzard\test.lua
Time: Thu Jun 28 13:42:57 2018
Count: 6
```

</p>
</details>

---

https://github.com/FailcoderAddons/supervillain-ui/blob/d05f090bb6de8a6d72a15b465688ccab30854508/SVUI_!Core/system/_docklets/raidleader.lua#L140
https://github.com/FailcoderAddons/supervillain-ui/blob/d05f090bb6de8a6d72a15b465688ccab30854508/SVUI_!Core/system/_docklets/raidleader.lua#L165

<details>
<summary>Vanilla WoW Lua Warning 2</summary><p>

```
Message: Frame SVUI_RaidToolMenu: Unknown script element OnClick
Time: Thu Jun 28 13:42:57 2018
Count: 1
```

</p>
</details>

**Comment by blueberryiswar** [comment]
I fixed the Aura things, as written in #200 they changed the API for Buffs and Debuffs (why the querries don't work anymore)

Artifacts are currently gone, so I would just uncomment the whole function. Same for onclick really.

Hope that helps :) Also, UNIT_POWER must be replaced with UNIT_POWER_UPDATE everywhere, see #199

**Comment by finalsliver** [comment]
@blueberryiswar Joey is currently working on a bunch of stuff in his own environment. not sure if you are on Discord but its the main social hub for us. https://discord.gg/guMxBMb

not sure what Joey has covered yet but it may be easier to get a response over tehre
